### PR TITLE
Remove use of MemoizeJac

### PR DESCRIPTION
**Pull request by nrontsis**
Removes use of `scipy.optimize.optimize.MemoizeJac` because it is not publicly exposed by recent versions of `scipy`. See https://github.com/mechmotum/cyipopt/issues/160 for more details. This is done by dropping support for passing `jac=True` in the scipy interface.

Testing: [The jax-scipy example was verified to work](https://github.com/mechmotum/cyipopt/blob/master/examples/hs071_scipy_jax.py).

### CODE REVIEW

**Comment by moorepants** [comment]
I looked a bit more closely here. See this test:

https://github.com/mechmotum/cyipopt/blob/master/cyipopt/tests/unit/test_scipy_optional.py#L164

If you provide a single function that calculates the objective and its gradient (similarly for the constraints), then you set `jac=True` and the function is cached so that the function and gradient are never calculated twice (because they are called at least twice for the same args).

The change you have here probably loses that functionality. This isn't well documented, but I think it is correct functionality to use `MemoizeJac` to avoid the duplicate computations.

**Comment by nrontsis** [comment]
Thanks, I agree that if a combined callback is used, then using some of sort of caching makes sense.

But do you think it’s worth maintaining support for combined callbacks at the expense of using an effectively nonpublic scipy import?

**Comment by moorepants** [comment]
I've opened a fix in scipy. The class was public at one point and we can see if they'll return it to being public.

**Comment by moorepants** [comment]
Thanks for the submission. I'm closing in favor of #183
